src/libs/scene_utils/client/cesium.ts

```diff
@@ -0,0 +1,400 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved. 2022
+// SPDX-License-Identifier: Apache-2.0
+
+import { createReadStream, createWriteStream } from 'fs';
+import { ArchiveCreationRequest, ModelType, OnComplete, UploadAssetRequest, UploadLocation } from '../cesium/types';
+import { Credentials, S3 } from 'aws-sdk';
+import { logProgress } from '../utils/file_utils';
+import { basename, join } from 'path';
+import fetch, { RequestInit, Response } from 'node-fetch';
+
+export class CesiumClient {
```

Comment:
Is there a reason for using a classical component rather than functional here? 

By using a classical component this code isn't able to utilize the React lifecycle hooks which impacts our ability to optimize our codebase. Per the react docs:

[_"We intend for Hooks to cover all existing use cases for classes, but we will keep supporting class components for the foreseeable future. At Facebook, we have tens of thousands of components written as classes, and we have absolutely no plans to rewrite them. Instead, we are starting to use Hooks in the new code side by side with classes."_](https://legacy.reactjs.org/docs/hooks-intro.html#gradual-adoption-strategy)

I'm happy to dive deeper into this with you if you'd like & am available for any questions you might have on how to convert this to a functional component. Doing so will not only allow you to take advantage of React's lifecycle management, but also removes unnecessary dependencies on external components.

Reply:
This is a good recommendation, but I don't think it's worth refactoring for the current use of this script. This will be a quick command line tool to get customers started with using 3D Tiles in TwinMaker. This isn't meant for production or any workflow on the browser yet. 
I will definitely keep this in mind when migrating this code to the AppKit in the future, or when we use something similar in our service Console.

Reply:
Looks like this is not in a React package. My feedback is based on that. You can disregard feedback for this review, but good to know when you're in our React packages. :D